apps/frontend/core-dapp/src/features/deposit/DepositStore.ts

```diff
@@ -106,17 +112,41 @@ export class DepositStore {
     return this.root.baseTokenStore.needToAllowFor(this.depositAmount, 'preCT')
   }
 
-  get globalNetDepositAmountInUsd(): string | undefined {
-    const depositAmount = this.root.depositRecordStore.globalNetDepositAmount
-    if (!depositAmount) return undefined
+  get globalDepositCapExceeded(): 'loading' | 'no' | 'already-exceeded' | 'exceeded-if-deposit' {
+    const { globalNetDepositAmount, globalNetDepositCap } = this.root.depositRecordStore
+    const { depositAmountBN } = this
 
```

Comment:
should we add a production check here so it doesn't affect the current behaviour?

```suggestion
    // TODO: remove this when contract is updated
    if (isProduction()) return 'no'
```